typed-racket-lib/typed-racket/base-env/base-types-extra.rkt

```diff
@@ -17,7 +17,7 @@
 (define-other-types
   -> ->* case-> U Rec All Opaque Vector
   Parameterof List List* Class Object Values Instance Refinement
-  pred Struct Struct-Type Prefab Top Bot)
+  pred Struct Struct-Type Prefab Top Bot Distinction)
```

Comment:
Oh it might be because I never allowed Distinction types to be subtypes of themselves.